src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -226,6 +240,43 @@ app.directive('dashboardApp', function ($injector) {
         $scope.indexPatterns = dashboardState.getPanelIndexPatterns();
       };
 
+      $scope.filter = function (field, value, operator, index) {
+        if ($scope.model.query.language === 'lucene') {
+          filterManager.add(field, value, operator, index);
+        }
+
+        if ($scope.model.query.language === 'kuery') {
+          const fieldName = _.isObject(field) ? field.name : field;
+          const kueryAST = fromKueryExpression($scope.model.query.query);
+          const newAST = addNode(
+            kueryAST,
+            nodeTypes.match.buildNode({ field: fieldName, values: value, operation: operator })
+          );
+          $scope.updateQuery({ query: toKueryExpression(newAST), language: 'kuery' });
+        }
```

Comment:
Similar to the comment below, much of this code will be extracted into a separate module in a follow up PR.